src/controllers/customer.js

```diff
@@ -43,6 +49,76 @@ const createCustomer = async (req, res) => {
   }
 }
 
+const getCustomers = async (req, res) => {
+  const customers = await getCustomersDb()
+  return res.json({ customers })
+}
+
+const getCustomerById = async (req, res) => {
+  const id = Number(req.params.id)
+
+  try {
+    const customer = await getCustomerByIdDb(id)
+    return res.json({ customer })
+  } catch (error) {
+    if (error instanceof PrismaClientKnownRequestError) {
+      if (error.code = "P2025") {
+        return res.status(404).json({ error: 'not found'})
+      }
+    }
+    return res.status(404)
+  }
+}
+
+const updateCustomer = async (req, res) => {
+  const id = Number(req.params.id)
+  const { name, phone, email } = req.body
+
+  if (!id) {
+    return res.json(400).json({
+      error: `Supplied id ${id} is not valid`
+    })
+  }
+
+  if (!name) {
+    return res.status(400).json({
+      error: 'Missing fields in request body'
+    })
+  }
+
+  try {
+    const customer = await updateCustomerNameDb(id, name)
+    
+    if (customer.contact) {
```

Comment:
There's a very unlikely edge-case here that if the `updateCustomerNameDb` function runs fine but, for whatever reason, the `customer` object does not have a `contact` property on it, this controller will do nothing and the client's request will time out.

It shouldn't ever happen, but if it did...